### PR TITLE
Implement special case for the delegatecall proxy pattern

### LINKED ISSUE #675: Add a special case for passing original calldata with symbolic length
## Description

As described in #674, Mythril currently doesn't support the delegatecall proxy pattern where the original calldata is passed through. This is because we don't support calldata with a symbolic offset or length.

I suppose that we could add a special case for `offset = 0` and `length = calldatasize` where we provide the original calldata to the callee.

### PR DESCRIPTION
**Pull request by nbanmp**
closes #675

### CODE REVIEW

File: mythril/laser/ethereum/call.py [hunk 1]
```diff
@@ -179,16 +180,26 @@ def get_call_data(
...
+            uses_entire_calldata = simplify(
+                memory_size - global_state.environment.calldata.calldatasize == 0
             )
-        ]
-        call_data = ConcreteCalldata(transaction_id, calldata_from_mem)
-        log.debug("Calldata: " + str(call_data))
+        except Z3Exception:
+            pass
+
+        if uses_entire_calldata:
```
Comment (norhh) [inline mythril/laser/ethereum/call.py]: does this part ever get executed?

File: mythril/laser/ethereum/call.py [hunk 2]
```diff
@@ -179,16 +180,26 @@ def get_call_data(
     )
 
     try:
-        calldata_from_mem = state.memory[
-            util.get_concrete_int(memory_start) : util.get_concrete_int(
-                memory_start + memory_size
+        uses_entire_calldata = False
+        try:
+            uses_entire_calldata = simplify(
```
Comment (norhh) [inline mythril/laser/ethereum/call.py]: is the "try" statement in line 182 needed to be covered for these lines? As i think TypeError should be caught from lines 194:197 and not from these lines right?

File: mythril/laser/ethereum/call.py [hunk 3]
```diff
@@ -179,16 +180,26 @@ def get_call_data(
...
+                memory_size - global_state.environment.calldata.calldatasize == 0
             )
-        ]
-        call_data = ConcreteCalldata(transaction_id, calldata_from_mem)
-        log.debug("Calldata: " + str(call_data))
+        except Z3Exception:
```
Comment (norhh) [inline mythril/laser/ethereum/call.py]: I can't get the reason why Z3Exception is caught here? can you explain.

Comment (nbanmp) [reply mythril/laser/ethereum/call.py]: :sweat_smile: A previous version had `if simplify(...): `, which would raise an exception. It's unneeded now.

**Pull request merged**